### PR TITLE
App name unique validation

### PR DESCRIPTION
**Pull request by nwmac**
This PR reworks the unique  name validator.

It can be used by reactive forms as well as template forms and the validator now makes the http call directly.

### CODE REVIEW

File: src/app/features/applications/app-name-unique.directive/app-name-unique.directive.ts [hunk 1]
```diff
@@ -1,46 +1,101 @@
...
+      return v ? null : {appNameTaken: true};
+    });
+  }
+
+  private checkAppName(cfGuid, spaceGuid, currentName, name): Observable<any> {
+    if (name === currentName || name.length === 0) {
+      return Observable.of(true);
     }
+    const options = new RequestOptions();
+    options.url = '/pp/v1/proxy/v2/apps';
```
Comment (richard-cox) [inline src/app/features/applications/app-name-unique.directive/app-name-unique.directive.ts]: The url should be constructed from the environment object (in `src/environments/environment.ts` as per api.efffect `/pp/${proxyAPIVersion}/proxy/${cfAPIVersion}`. If we do more of these we should pull this a lot of this section into something generic

File: src/app/features/applications/app-name-unique.directive/app-name-unique.directive.ts [hunk 2]
```diff
@@ -1,46 +1,101 @@
...
+    options.params = new URLSearchParams('');
+    options.params.set('q', 'name:' + name);
+    options.params.append('q', 'space_guid:' + spaceGuid);
+    options.method = 'get';
+    options.headers = new Headers();
+    options.headers.set('x-cap-cnsi-list', cfGuid);
+    options.headers.set('x-cap-passthrough', 'true');
+    return this.http.request(new Request(options)).map(response => {
```
Comment (richard-cox) [inline src/app/features/applications/app-name-unique.directive/app-name-unique.directive.ts]: We should silently handle errors here i guess? Not sure what kind of error the http library throws out and if they need to be swallowed in a catch on the observable

File: src/app/features/applications/app-name-unique.directive/app-name-unique.directive.ts (hunk 1 above)
Comment (nwmac) [reply src/app/features/applications/app-name-unique.directive/app-name-unique.directive.ts]: Done.

File: src/app/features/applications/app-name-unique.directive/app-name-unique.directive.ts (hunk 2 above)
Comment (nwmac) [reply src/app/features/applications/app-name-unique.directive/app-name-unique.directive.ts]: I have added error handling

**Pull request merged**